docs/solidity-by-example.rst

```diff
@@ -645,4 +645,499 @@ Safe Remote Purchase
 Micropayment Channel
 ********************
 
-To be written.
+Creating and verifying signatures
+=================================
+
+Signatures are used to authorize transactions,
+and they're a general tool that's available to
+smart contracts. In this chapter we'll use it to
+prove to a smart contract that a certain account
+approved a certain message. We'll build a simple
+smart contract that lets me transmit ether, but
+in a unsual way, instead of calling a function myself
+to initiate a payment, I'll let the receiver of
+the payment do that, and therefore pay the transaction
+fee. All this using cryptographics signatures that I'll
+make and send off-chain (e.g. via email). We'll call our
+contract as ReceiverPays, it will works a lot like
+writing a check.Our contract will works like that:
+
+    1. The owner deploys ReceiverPays, attaching enough ether to cover the payments that will be made.
+    2. The owner authorizes a payment by signing a message with their private key.
+    3. The owner sends the cryptographically signed message to the designated recipient. The message does not need to be kept secret (you'll understand it later), and the mechanism for sending it doesn't matter.
+    4. The recipient claims their payment by presenting the signed message to the smart contract, it verifies the authenticity of the message and then releases the funds.
+
+Creating the signature
+----------------------
+
+We don't need the to interact with Ethereum network to
+do that. The proccess is completely offline, and the
+every major programming language has the libraries to do it.
+The signature algorithm Ethereum has the support for is the
+`Elliptic Curve Signature Algorithm(EDCSA) <https://en.wikipedia.org/wiki/Elliptic_Curve_Digital_Signature_Algorithm>`_.
+In this tutorial, we'll signing messages in the web browser
+using web3.js and MetaMask. We can sign messages in too many ways,
```

Comment:
If we write our own contract to verify, we do not need to take care about compatibility anyway, right? Perhaps what we should go a little bit into detail here is that we have to take care that nobody can use the signed message for a different purpose.